src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const stateHistory = [];
+  let currentState = {};
+
+  Object.assign(currentState, state);
+
+  for (const { type, extraData, keysToRemove } of actions) {
+    switch (type) {
+      case 'addProperties':
+        Object.assign(currentState, extraData);
+        break;
+
+      case 'removeProperties':
+        for (const i of keysToRemove) {
+          delete currentState[i];
+        }
+        break;
 
+      case 'clear':
+        currentState = {};
+        break;
+
+      default:
+        throw new Error();
+    }
```

Comment:
Adding an informative message to throw new Error() in the default case.
```suggestion
      default:
        throw new Error(`Unexpected action type: ${type}`);
    }
```